### PR TITLE
fixed ads going out of bounds

### PR DESCRIPTION
**Pull request by shnartho**
Fix: ads going out of bounds

### CODE REVIEW

File: src/components/ads/AdHandling.js [hunk 1]
```diff
@@ -63,26 +62,11 @@ const AdHandler = () => {
...
-    const currentAds = ads.slice(indexOfFirstAd, indexOfLastAd);
-
-    const totalPages = Math.ceil(ads.length / adsPerPage);
-
-    const goToNextPage = () => {
-        setCurrentPage((prevPage) => Math.min(prevPage + 1, totalPages));
-    };
-
-    const goToPreviousPage = () => {
-        setCurrentPage((prevPage) => Math.max(prevPage - 1, 1));
```
Comment (shnartho) [inline src/components/ads/AdHandling.js:78]: ok

File: src/components/ads/AdHandling.js [hunk 2]
```diff
@@ -13,8 +12,8 @@ const AdHandler = () => {
             try {
                 const response = await fetch("https://metasurfai-public-api.fly.dev/v1");
                 const data = await response.json();
-                console.log(data);
```
Comment (shnartho) [inline src/components/ads/AdHandling.js:16]: 👍 `console.log()` need to be removed from everywhere for prod codes.

**Comment by JemalSmb** [comment]
fixed most issues, added aspect-ratio logic. can't view preview so can you green light it so I close the pr

File: src/styles/styles.css [hunk 3]
```diff
@@ -36,26 +38,68 @@ body {
...
     .ads-container {
-        grid-template-columns: repeat(1, 1fr);
+        grid-template-columns: repeat(1, 1fr); /* 1 column */
     }
-}
\ No newline at end of file
+}
+
+
+/* For horizontal ads (16:9 aspect ratio) */
```
Comment (JemalSmb) [inline src/styles/styles.css:66]: this is aspect-ratio logic, where if the image gets formalized as 16:9 or 9:16 ratio

File: src/components/ads/AdHandling.js [hunk 4]
```diff
@@ -125,26 +121,7 @@ const AdHandler = () => {
                     </div>
                 </div>
             )}
-             {/* Pagination Controls */}
```
Comment (JemalSmb) [inline src/components/ads/AdHandling.js:128]: removed pagination per request

File: src/components/ads/AdHandling.js [hunk 5]
```diff
@@ -102,16 +98,16 @@ const AdHandler = () => {
             </div>
 
             {selectedAd && (
-                <div className="popup-ads fixed top-0 left-0 z-50 w-full h-full bg-black bg-opacity-80 flex items-center justify-center">
-                    <div className="relative bg-black bg-opacity-50 p-4 rounded-lg border-2 border-opacity-40 border-pink-600">
-                        <span onClick={closeModal} className="absolute top-2 right-2 text-3xl font-bold cursor-pointer">
+                <div className="p-4 fixed top-0 left-0 z-50 w-full h-full bg-black bg-opacity-80 flex items-center justify-center">
```
Comment (JemalSmb) [inline src/components/ads/AdHandling.js:101]: changed some tailwind css to insure it looks user friendly on all types of screes (mobile, tablet and desktop)

File: src/components/ads/AdHandling.js [hunk 6]
```diff
@@ -63,28 +62,25 @@ const AdHandler = () => {
         }
     };
 
-    // Pagination logic
-    const indexOfLastAd = currentPage * adsPerPage;
-    const indexOfFirstAd = indexOfLastAd - adsPerPage;
-    const currentAds = ads.slice(indexOfFirstAd, indexOfLastAd);
+    // Determine the aspect ratio class
```
Comment (JemalSmb) [inline src/components/ads/AdHandling.js:65]: aspect ratio logic

**Comment by shnartho** [comment]
> fixed most issues, added aspect-ratio logic. can't view preview so can you green light it so I close the pr

We can not preview it in this branch, we need to check the code, if everything looks okay we need to merge it into main branch which will trigger the deployment.

File: src/components/ads/AdHandling.js [hunk 7]
```diff
@@ -13,8 +12,8 @@ const AdHandler = () => {
             try {
                 const response = await fetch("https://metasurfai-public-api.fly.dev/v1");
                 const data = await response.json();
-                console.log(data);
-                setAds(data);
+                const sortedAds = data.sort((a, b) => b.token_reward - a.token_reward); // Sort by token_reward descending
```
Comment (JemalSmb) [inline src/components/ads/AdHandling.js:15]: sorting ads based on token reward logic

File: src/components/ads/AdHandling.js (hunk 4 above)
Comment (shnartho) [reply src/components/ads/AdHandling.js:128]: Perfect, we will have more user friendly UI

**Review by JemalSmb — APPROVED** [review]
these changes fixed the ads going out of bound. Added aspect-ratio logic to the ads viewing.

**Pull request merged**